Bio/PDB/cealign/ccealignmodule.cpp

```diff
@@ -0,0 +1,747 @@
+    return NULL;
+  }
+
+  PyObject* pyU = Py_BuildValue( "[f,f,f,f, f,f,f,f, f,f,f,f, f,f,f,f]",
+				 bestU[0][0], bestU[1][0], bestU[2][0], bestCOM1[0],
+				 bestU[0][1], bestU[1][1], bestU[2][1], bestCOM1[1],
+				 bestU[0][2], bestU[1][2], bestU[2][2], bestCOM1[2],
+				 -bestCOM2[0], -bestCOM2[1], -bestCOM2[2], 1.);
+
+  PyObject* pyPathA = PyList_New(0);
+  PyObject* pyPathB = PyList_New(0);
+  for (int j = 0; j < smaller && paths[bestO][j].first != -1; j++) {
+    PyObject* v = Py_BuildValue("i", paths[bestO][j].first);
+    PyList_Append(pyPathA, v);
+    Py_DECREF(v);
+    v = Py_BuildValue("i", paths[bestO][j].second);
+    PyList_Append(pyPathB, v);
+    Py_DECREF(v);
+  }
+
+  return Py_BuildValue("[ifNNN]", bestLen, bestRMSD, pyU, pyPathA, pyPathB);
+}
+
+
+TA2<double> transpose(const TA2<double>& v)
+{
+  int m = (int) v.dim1();
+  int n = (int) v.dim2();
+	
+  TA2<double> rVal(n,m);
+	
+  for ( int i = 0; i < m; i++ )
+    for ( int j = 0; j < n; j++ )
+      rVal[j][i] = v[i][j];
+		
+  return rVal;
+}
+
+// Main Functions
+PyObject * ExecutiveCEAlign(PyObject * listA, PyObject * listB, int lenA, int lenB, float d0, float d1, int windowSize, int gapMax) {
+  int i=0;
+  int smaller;
+  double **dmA ,**dmB, **S;
+  int bufferSize;
+  pcePoint coordsA, coordsB;
+  pathCache paths = NULL;
+  PyObject * result;
+
+  smaller = lenA < lenB ? lenA : lenB;
+  /* get the coodinates from the Python objects */
+  coordsA = (pcePoint) getCoords(listA, lenA);
+  coordsB = (pcePoint) getCoords(listB, lenB);
+	
+  /* calculate the distance matrix for each protein */
+  dmA = (double**) calcDM(coordsA, lenA);
+  dmB = (double**) calcDM(coordsB, lenB);
+	
+  /* calculate the CE Similarity matrix */
+  S = (double**) calcS(dmA, dmB, lenA, lenB, windowSize);
+	
+  /* find the best path through the CE Sim. matrix */
+  bufferSize = 0;
+
+  /* the following line HANGS PyMOL */
+  paths = (pathCache) findPath(S, dmA, dmB, lenA, lenB, d0, d1, windowSize, gapMax, &bufferSize);
+	
+  /* Get the optimal superposition here... */
+  result = (PyObject*) findBest(coordsA, coordsB, paths, bufferSize, smaller, windowSize);
+	
+  /* release memory */
+  free(coordsA);
+  free(coordsB);
+  for ( i = 0; i < bufferSize; ++i )
+    free(paths[i]);
+  free(paths);
+
+  /* distance matrices	 */
+  for ( i = 0; i < lenA; i++ )
+    free( dmA[i] );
+  free(dmA);
+
+  for  ( i = 0; i < lenB; i++ )
+    free( dmB[i] );
+  free(dmB);
+
+  /* similarity matrix */
+  for ( i = 0; i < lenA; i++ )
+    free( S[i] );
+  free(S);
+
+  return (PyObject*) result;
+}
+
+//
+// Python Interface
+//
+static PyObject* run_cealign(PyObject *self, PyObject *args)
+{
+  int windowSize = 8, gap_max = 30;
+  float d0 = 3.0, d1 = 4.0;
+  PyObject *listA, *listB, *result;
+  Py_ssize_t lenA, lenB;
+
+  /* Unpack the arguments from Python */
+  if ( ! PyArg_ParseTuple(args, "OO|ii", &listA, &listB, &windowSize, &gap_max) ){
+      printf("Error parsing arguments for run_cealign\n");
```

Comment:
This line can be removed; PyArg_ParseTUple will raise an exception if something goes wrong.